source/common/config/delta_subscription_state.cc

```diff
@@ -138,10 +177,14 @@ void DeltaSubscriptionState::handleGoodResponse(
   //
   // So, leave the version map entry present but blank. It will be left out of
   // initial_resource_versions messages, but will remind us to explicitly tell the server "I'm
-  // cancelling my subscription" when we lose interest.
+  // cancelling my subscription" when we lose interest. In case of resources received as a part of
+  // the wildcard subscription, we just drop them.
   for (const auto& resource_name : message.removed_resources()) {
-    if (resource_names_.find(resource_name) != resource_names_.end()) {
-      setResourceWaitingForServer(resource_name);
+    if (auto maybe_resource = getRequestedResourceState(resource_name);
+        maybe_resource.has_value()) {
+      maybe_resource->setAsWaitingForServer();
+    } else {
+      wildcard_resource_state_.erase(resource_name);
```

Comment:
Would it be possible to add some invariants (ASSERTs) showing that wildcard resource state and request resource state are disjoint?

Reply:
Sure, I can add them.